sdk/android/azure-android-compat/src/androidTest/java/com/samples/androidcompat/AzCoreReflectionInstrumentedTests.java

```diff
@@ -0,0 +1,95 @@
+package com.samples.androidcompat;
+
+//import static com.azure.core.implementation.ReflectionUtils.getConstructorInvoker;
+//import com.azure.core.implementation.ReflectionUtils;
+import com.azure.ai.openai.OpenAIClient;
+import com.azure.ai.openai.OpenAIClientBuilder;
+import com.azure.ai.translation.text.TextTranslationClient;
+import com.azure.ai.translation.text.TextTranslationClientBuilder;
+import com.azure.core.credential.AzureKeyCredential;
+
+import com.azure.identity.ClientSecretCredentialBuilder;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.*;
+
+/**
+ * Instrumented test cases for ReflectionUtils usages.
+ * When an emulator's API and an app's targetSdk are both greater than 27, ExceptionInInitializer
+ * would occur because of the constructor for MethodHandles.Lookup being missing. These three test
+ * cases replicated user-reported cases where ExceptionInInitializer would occur, and were used to
+ * test if a new PR could resolve these issues by providing two different ReflectionUtil versions.
+ */
+@RunWith(AndroidJUnit4.class)
+public class AzCoreReflectionInstrumentedTests {
+    // Tests were run with placeholder credentials but did trigger ExceptionInInitializerError.
+    /* Testing issue 1540 at https://github.com/microsoftgraph/msgraph-sdk-java/issues/1540.
+     * Results for current azure-core version (1.42.0):
+     * - API level = 26, 27: IllegalArgumentException (invalid TENANT_ID)
+     * - API level > 27: ExceptionInInitializerError (MethodHandles.Lookup)
+     *
+     * Results for PR version:
+     * - API level >= 26: IllegalArgumentException (invalid TENANT_ID)
+     * This indicates that the PR has managed to resolve the ExceptionInInitializerError.
+     */
+    @Test
+    public void identityTest() {
+        // Context of the app under test.
+        String TENANT_ID = "<redacted>";
+        String CLIENT_ID = "<redacted>";
+        String CLIENT_SECRET = "<redacted>";
```

Comment:
Appreciate that this is redacted so it isn't part of the github history, but how does this test run when running from a command line interface? Are you able to bring the values in from the environment, because no one will be running these tests manually.

Reply:
I just used the placeholders for the instrumented tests without using actual credentials, since the ExceptionInInitializerErrors from the three issues would still get thrown before the credentials get properly used. As a result, I wasn't sure whether the tests still needed to be run with valid credentials. I'll see if Saya can add API 30 and 34 emulators to the CI, that should demonstrate the ExceptionInInitializerErrors.

I'll act today on the feedback you've given for the naming conventions, tests and dependencies of azure-android-compat, I agree that a lot of it was unnecessary, so I'll clean it up and our team should be able to decide on the app consolidation in a group meeting this week.